### PR TITLE
[14.0] [mig] hr_holidays_public

### PR DESCRIPTION
**Pull request by dsolanki-initos**

### CODE REVIEW

**Review by aphon61bank — APPROVED** [review]
code review.

**Review by Yoshiro009 — APPROVED** [review]

**Comment by pedrobaeza** [comment]
Please include #20

**Comment by fshah-initos** [comment]
@pedrobaeza I have included https://github.com/OCA/hr-holidays/pull/20 to migrated v14. Please review.
Thanks in advance :)

**Comment by pedrobaeza** [comment]
Well, you should cherry-pick the commit for preserving attribution instead of doing on top of it. You didn't need to bump version number as well.

File: hr_holidays_public/readme/CONFIGURE.rst [hunk 1]
```diff
@@ -0,0 +1,3 @@
+Go to *Leaves -> Configuration* and open a Leave Type
```
Comment (emagdalenaC2i) [inline hr_holidays_public/readme/CONFIGURE.rst]:
```suggestion
Go to *Time Off -> Configuration -> Time Off Types* and open a Leave Type
```

File: hr_holidays_public/readme/USAGE.rst [hunk 2]
```diff
@@ -0,0 +1,17 @@
+For adding public holidays:
+
+#. Go to the menu *Leaves > Public Holidays > Public Holidays*.
```
Comment (emagdalenaC2i) [inline hr_holidays_public/readme/USAGE.rst:3]:
```suggestion
#. Go to the menu *Time Off -> Public Holidays -> Public Holidays*.
```

File: hr_holidays_public/readme/USAGE.rst [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+#. Create your public holidays.
+
+For using public holidays on leaves:
+
+#. Go to *Leaves > Dashboard*.
```
Comment (emagdalenaC2i) [inline hr_holidays_public/readme/USAGE.rst]:
```suggestion
#. Go to *Time Off -> Dashboard*.
```

**Review by emagdalenaC2i — COMMENTED** [review]
Tested in runbot, LGTM. Just a few minor changes, attend pending comments and squash your commits

File: hr_holidays_public/__manifest__.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
+# Copyright 2015 2011,2013 Michael Telahun Makonnen <mmakonnen@gmail.com>
+# Copyright 2020 InitOS Gmbh
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+{
+    "name": "HR Holidays Public",
+    "version": "14.0.3.0.1",
```
Comment (emagdalenaC2i) [inline hr_holidays_public/__manifest__.py]:
```suggestion
    "version": "14.0.1.0.0",
```

**Comment by fshah-initos** [comment]
@emagdalenaC2i Suggested changes are performed.
Kindly review.

**Review by emagdalenaC2i — APPROVED** [review]

**Comment by fshah-initos** [comment]
@pedrobaeza Suggested changes are approved.It seems this branch is ready to get merged with 14.
Please have a look.

Kind Regards,

File: hr_holidays_public/__manifest__.py [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    "name": "HR Holidays Public",
+    "version": "14.0.1.0.0",
+    "license": "AGPL-3",
+    "category": "Human Resources",
+    "author": "Michael Telahun Makonnen, "
+    "Tecnativa, "
+    "Fekete Mihai (Forest and Biomass Services Romania), "
+    "Druidoo, "
+    "Odoo Community Association (OCA),"
+    "InitOS Gmbh",
```
Comment (pedrobaeza) [inline hr_holidays_public/__manifest__.py]: You can't add this for a simple migration. Please remove it.

File: hr_holidays_public/migrations/14.0.1.0.0/post-migration.py [hunk 6]
```diff
@@ -0,0 +1,22 @@
+# Copyright 2019 Druidoo - Iván Todorovich <ivan.todorovich@druidoo.io>
```
Comment (pedrobaeza) [inline hr_holidays_public/migrations/14.0.1.0.0/post-migration.py]: This file must be removed, not renamed.

File: hr_holidays_public/models/hr_leave.py [hunk 7]
```diff
@@ -0,0 +1,21 @@
+# Copyright 2017-2018 Tecnativa - Pedro M. Baeza
+# Copyright 2018 Brainbean Apps
+# Copyright 2020 InitOS Gmbh
```
Comment (pedrobaeza) [inline hr_holidays_public/models/hr_leave.py]: You mustn't add any copyright to a file that you don't touch (undo it for all the untouched files).

File: hr_holidays_public/security/ir.model.access.csv [hunk 8]
```diff
@@ -0,0 +1,8 @@
+id,name,model_id:id,group_id:id,perm_read,perm_write,perm_create,perm_unlink
+access_hr_holidays_public_user,access_hr_holidays_public,model_hr_holidays_public,base.group_user,1,0,0,0
+access_hr_holidays_public_manager,access_hr_holidays_public,model_hr_holidays_public,hr_holidays.group_hr_holidays_manager,1,1,1,1
+access_hr_holidays_public_line_user,access_hr_holidays_public_line,model_hr_holidays_public_line,base.group_user,1,0,0,0
+access_hr_holidays_public_line_manager,access_hr_holidays_public_line,model_hr_holidays_public_line,hr_holidays.group_hr_holidays_manager,1,1,1,1
+access_hr_holidays_public_manager_next_year_wizard,access_hr_holidays_public_next_year_wizard,model_public_holidays_next_year_wizard,base.group_user,1,1,1,0
```
Comment (pedrobaeza) [inline hr_holidays_public/security/ir.model.access.csv]: This should be adjusted to the group that is allowed to access to such wizard, not all. I also wonder why don't you grant unlink permission.

File: hr_holidays_public/security/ir.model.access.csv [hunk 9]
```diff
@@ -0,0 +1,8 @@
...
+access_hr_holidays_public_manager_user,access_hr_holidays_public,model_hr_holidays_public,base.group_user,1,1,1,0
```
Comment (pedrobaeza) [inline hr_holidays_public/security/ir.model.access.csv]: Why are you giving such broad permission to all the users? In fact, the first line already grants the needed one.

File: hr_holidays_public/security/ir.model.access.csv [hunk 10]
```diff
@@ -0,0 +1,8 @@
...
+access_hr_leaves,access_hr_leaves_public,model_hr_leave,base.group_user,1,1,1,0
```
Comment (pedrobaeza) [inline hr_holidays_public/security/ir.model.access.csv]: This is totally wrong. Remove this.

**Review by pedrobaeza — DISMISSED** [review]

**Comment by fshah-initos** [comment]
@pedrobaeza Is it possible to merge?

**Comment by pedrobaeza** [comment]
Please squash your changes and leave @victoralmau's commit with only his changes, as you have included the changes for the review in it.

**Comment by fshah-initos** [comment]
@pedrobaeza I have squash the commit.You can now merge it.

**Comment by pedrobaeza** [comment]
Selección_010

**Comment by fshah-initos** [comment]
@pedrobaeza I have squash this two commits. For the lost attribution commit I have kept it seperate.

**Pull request merged**